crates/router/src/connector/esnekpos/transformers.rs

```diff
@@ -0,0 +1,385 @@
+                                                (Some(line1), Some(line2)) => {
+                                                    let line1val = format!("{:?}", line1);
+                                                    let line2val = format!("{:?}", line2);
+
+                                                    let addstr =
+                                                        format!("{}, {}", line1val, line2val);
+                                                    Some(Secret::new(addstr))
+                                                }
+                                                (Some(line1), None) => Some(line1),
+                                                (None, Some(line2)) => Some(line2),
+                                                (None, None) => None,
+                                            }
+                                        }
+                                        None => None,
+                                    },
+                                    client_ip: req.get_ip_address_as_optional().clone(),
+                                });
+                                let customer = match customer_res {
+                                    Ok(customer) => customer,
+                                    Err(e) => return Err(e.into()),
+                                };
+
+                                let auth = EsnekposAuthType::try_from(
+                                    &item.router_data.connector_auth_type,
+                                )?;
+
+                                Ok(Self {
+                                    config: EsnekposPaymentRequestConfig {
+                                        merchant: auth.merchant.expose(),
+                                        merchant_key: auth.merchant_key.expose(),
+                                        back_url: item.router_data.request.get_return_url()?,
+                                        order_ref_number: item.router_data.payment_id.clone(),
+                                        prices_currency: item
+                                            .router_data
+                                            .request
+                                            .currency
+                                            .to_string(),
+                                        order_amount: item.amount.get_amount_as_string(),
+                                    },
+                                    credit_card,
+                                    customer,
+                                    product: vec![],
+                                })
+                            }
+                        }
+                    }
+                    _ => Err(
+                        errors::ConnectorError::NotImplemented("Payment methods".to_string())
+                            .into(),
+                    ),
+                }
+            }
+            Err(e) => Err(e.into()),
+        }
+    }
+}
```

Comment:
Please don't nest the match statements, it reduces the readability of code, rather try to extract the data that are needed separately and and then construct the struct. For help, you can use utils function present in `crates/router/src/connector/utils.rs` for such extraction of data. Please go through any connector implementation as an example to construct in the following way. For example, you can go through try_from implementation for `AdyenPaymentRequest` in `crates/router/src/connector/adyen/transformers.rs` for an understanding.